trade-finance-manager-api/src/v1/controllers/amendment.controller.js

```diff
@@ -147,56 +113,59 @@ const getAmendmentById = async (req, res) => {
 };
 
 const getAmendmentByFacilityId = async (req, res) => {
-  const { facilityId } = req.params;
-  const amendment = await api.getAmendmentByFacilityId(facilityId);
+  const { facilityId, amendmentIdOrStatus, type } = req.params;
+  let amendment;
+  if (amendmentIdOrStatus === CONSTANTS.AMENDMENTS.AMENDMENT_QUERY_STATUSES.IN_PROGRESS) {
+    amendment = (await api.getAmendmentInProgress(facilityId)).data;
+  } else if (amendmentIdOrStatus === CONSTANTS.AMENDMENTS.AMENDMENT_QUERY_STATUSES.COMPLETED) {
+    if (type === CONSTANTS.AMENDMENTS.AMENDMENT_QUERIES.LATEST_COVER_END_DATE) {
+      amendment = await api.getLatestCompletedAmendmentDate(facilityId);
+    } else if (type === CONSTANTS.AMENDMENTS.AMENDMENT_QUERIES.LATEST_VALUE) {
+      amendment = await api.getLatestCompletedAmendmentValue(facilityId);
+    } else {
+      amendment = await api.getCompletedAmendment(facilityId);
+    }
+  } else if (ObjectId.isValid(amendmentIdOrStatus)) {
+    amendment = await api.getAmendmentById(facilityId, amendmentIdOrStatus);
+  } else if (!amendmentIdOrStatus && !type) {
+    amendment = await api.getAmendmentByFacilityId(facilityId);
+  }
   if (amendment) {
     return res.status(200).send(amendment);
   }
   return res.status(422).send({ data: 'Unable to get the amendment by facilityId' });
 };
 
 const getAmendmentsByDealId = async (req, res) => {
-  const { dealId } = req.params;
-  const amendment = await api.getAmendmentsByDealId(dealId);
-  if (amendment) {
-    return res.status(200).send(amendment);
-  }
-  return res.status(422).send({ data: 'Unable to get the amendment by deal Id' });
-};
-
-const getAmendmentInProgressByDealId = async (req, res) => {
-  const { dealId } = req.params;
-  const amendment = await api.getAmendmentInProgressByDealId(dealId);
-  if (amendment) {
-    return res.status(200).send(amendment);
+  const { dealId, status, type } = req.params;
+  let amendment;
+  if (status === CONSTANTS.AMENDMENTS.AMENDMENT_QUERY_STATUSES.IN_PROGRESS) {
+    amendment = await api.getAmendmentInProgressByDealId(dealId);
+  } else if (status === CONSTANTS.AMENDMENTS.AMENDMENT_QUERY_STATUSES.COMPLETED) {
+    if (type === CONSTANTS.AMENDMENTS.AMENDMENT_QUERIES.LATEST) {
```

Comment:
Same as above, might be prudent to use independent `if` with compound conditions or `switch` or helper functions?

Reply:
Done with a switch, on the tfm-get-amendments controller as well